src/coreclr/jit/emitxarch.cpp

```diff
@@ -34,11 +34,22 @@ bool emitter::IsSSEOrAVXInstruction(instruction ins)
     return (ins >= INS_FIRST_SSE_INSTRUCTION) && (ins <= INS_LAST_AVX_INSTRUCTION);
 }
 
+bool emitter::IsSSEOrAVXorAVX512Instruction(instruction ins)
+{
+    // TODO-XArch-AVX512: Fix check once AVX512 instructions are added.
+    return (ins >= INS_FIRST_SSE_INSTRUCTION) && (ins <= INS_LAST_AVX_INSTRUCTION);
```

Comment:
Might be good to just pre-emptively define `INS_LAST_AVX512_INSTRUCTION` and check that here.

That also gives a "well-defined" spot to insert new instructions in the instrs header.

Reply:
Added INS_FIRST_AVX512_INSTRUCTION  and INS_LAST_AVX512_INSTRUCTION and added the suggested functions.